### ISSUE TITLE
BluePrint configure resourceGroup tags from parameter

### ISSUE THREAD

**Issue by pkhabazi** [Azure/azure-blueprints#52]
I am trying to configure tags for ResourceGroups from the blueprints and I am experiencing some issue.

In the example below I am want to parse the Tags as an object parameter:
```JSON
{
  "properties": {
    "type": "Microsoft.Blueprint/blueprints",
    "description": "Management Blueprint",
    "targetScope": "subscription",
    "parameters": {
      "tags": {
        "type": "object",
        "metadata": {
          "displayName": "Enter the Tags that need to be configured"
        },
        "defaultValue": {
          "tags1": "value1",
          "tags2": "value2"
        }
      }
    },
    "resourceGroups": {
      "ResourceGroup1": {
        "name": "resourceGroup01",
        "location": "westeurope",
        "metadata": {
          "displayName": "resourceGroup01"
        },
        "dependsOn": [],
        "tags": "[parameters('tags')]"
      }
    }
  }
}
```

However, the above template returns the error message below:
```PowerShell
Import-AzBlueprintWithArtifact: Can't deserialize the JSON file '/management/Blueprint.json'.  'Error converting value "[parameters('tags')]" to type 'System.Collections.Generic.IDictionary`2[System.String,System.String]'. Path 'tags', line 26, position 38.'
```

When I configure the Tags on ResourceGroup and use a String parameter as input it works fine:
```JSON
{
    "properties": {
      "type": "Microsoft.Blueprint/blueprints" ,
        "description": "Management Blueprint",
        "targetScope": "subscription",
        "parameters": {
          "tagsvalue": {
            "type": "string",
            "defaultValue": "value1"
          }
        
        },
        "resourceGroups": {
              "ResourceGroup1": {
                "name": "resourceGroup01",
                "location": "westeurope",
                "metadata": {
                  "displayName": "resourceGroup01"
                },
                "dependsOn": [],
                "tags": {
                  "Tags1": "[parameters('tagsvalue')]"
                }
              }
        }
    }
  }
```

Is this a bug or is it not possible to forward an object to the Tags part?

**Comment by DennisR73** [comment]
+1

**Comment by alex-frankel** [comment]
I don't think we support object parameters. I think the tag values need to be passed individually. Have you tried that?

**Comment by felipebbc** [comment]
Hi  @pkhabazi 
The reason I believe you are getting the error message '_Error converting value "[parameters('tags')]" to type 'System.Collections.Generic.IDictionary`2[System.String,System.String]_' is because the "[parameters('tags')]" expression within the resourceGroups section does not actually get substituted with the "tags" object you defined in the parameters section. The ARM interpreter is passing a literal string _"[parameters('tags')]"_ to the function that attempts conversion to a System.Collections.Generic.IDictionary<String,String> object and that is why it fails. 

So in your first example, you are actually trying to pass a string to the tags element while it is expecting an object. In your second example, you are passing an object. Keep in mind that in the second example, the "[parameters('tagsvalue')]" string is also not getting replaced with _"value1"_ when you run the Import-AzBlueprintWithArtifact command. It just sets the value of the "Tags1" key to the _"[parameters('tagsvalue')]"_ string. 

VS Code's syntax highlighting for ARM templates is misleading in this case. The color coding for the "[parameters('tags')]" expression in the resourceGroups section should be represented in the standard red denoting a string.

**Comment by torreymicrosoft** [comment]
Thanks for reporting that passing resource group tags as an object parameter fails to deserialize, while a string parameter works.

Heads-up on some important news: Azure Blueprints is being retired on January 31, 2027, with a phased wind-down starting July 31, 2026. After retirement the API, Azure CLI, PowerShell, and portal experience go away, and any definitions, versions, or assignments you have not exported are deleted (resources already deployed are not removed). Because of this, we are no longer shipping fixes or new features for Azure Blueprints.

The recommended replacement is Azure Deployment Stacks, which provides resource grouping, full lifecycle management, and deny assignment (lock) protection equivalent to blueprint locks. You can also publish your definitions as Template Specs or store them in Git for versioning.

Full details and the phased timeline are here:
- Announcement: https://aka.ms/AzureBlueprintsRetirementAnnouncement
- Migration guidance: https://aka.ms/AzureBlueprintsMigration
- FAQ: https://aka.ms/AzureBlueprintsRetirementFAQ

We are closing this issue as part of the Azure Blueprints retirement. If you run into anything while moving to Deployment Stacks or Template Specs, please open an issue in that repository so the right team can help. Thanks for using Azure Blueprints.
